vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -1,47 +1,35 @@
 ---
-  - name: Get the name of a worker bmh name
-    shell: |
-       kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
-       |select(.status.provisioning.state | contains("provisioned")) |select(.spec.consumerRef.name
-       | contains("{{ CLUSTER_NAME }}-workers")) | .metadata.name'
-    register: worker_bmh_name
-
-  - name: Get the names of master bmh name
-    shell: |
-       kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
-       |select(.status.provisioning.state | contains("provisioned")) |select(.spec.consumerRef.name
-       | contains("{{ CLUSTER_NAME }}-controlplane")) | .metadata.name'
-    register: master_bmh_name
-
-  - set_fact:
-      WORKER_BMH: "{{ worker_bmh_name.stdout }}"
-      MASTER_BMH_0: "{{ master_bmh_name.stdout_lines.0 }}"
-      MASTER_BMH_1: "{{ master_bmh_name.stdout_lines.1 }}"
-      MASTER_BMH_2: "{{ master_bmh_name.stdout_lines.2 }}"
-  
+  - name: Get the master objects
+    shell: "kubectl get m3m -n {{ NAMESPACE }} -o json |  jq -r '.items[] | \
+            select(.metadata.name | contains(\"controlplane\")) | \
+            (.metadata.annotations.\"metal3.io/BareMetalHost\"),(.metadata.ownerReferences |.[] | \
+            select(.kind==\"Machine\") | .name)'"
+    register: masters
+
+  - name: Get the worker objects
+    shell: "kubectl get m3m -n {{ NAMESPACE }} -o json | jq -r '.items[] | \
+            select(.metadata.name | contains(\"workers\")) | \
+            (.metadata.annotations.\"metal3.io/BareMetalHost\"),(.metadata.ownerReferences |.[] | \
+            select(.kind==\"Machine\") | .name)'"
```

Comment:
you should probably select the first element of the list to make sure that even if there are multiple Machine owner refs, you only get one ?

Reply:
Is it possible to have more than one Machine owner refs for a single m3m?

Reply:
yes, owner references is a list, so you can have multiple of the same type, referencing different objects.

Reply:
Fixed based on the discussion. Thank you.